core/wiki/macros/toc.tid

```diff
@@ -2,182 +2,212 @@ title: $:/core/macros/toc
 tags: $:/tags/Macro
 
 \define toc-caption()
-<$set name="tv-wikilinks" value="no">
-  <$transclude field="caption">
-    <$view field="title"/>
-  </$transclude>
-</$set>
+\whitespace trim
+<$set name="tv-wikilinks" value=no>
+  <$list filter="[all[current]has[caption]]" 
```

Comment:
I assume that this is a "fix" for blank caption fields? I think this isn't the right way to approach it. Those tiddlers with a blank caption are broken, and we should provide the tools to fix them. If we attempt to change the semantics of having a blank caption field then it's a disaster to do it inconsistently; it makes the overall system much more complex, with the only benefit being that certain mistakes can be "masked" in certain situations.

Reply:
This is the discussion why I did implement it: Table of Contents toc-caption macro awkward behaviour #3624 

The reason, why I thought it should be implemented is: 

 - Open tiddlywiki.com 
 - Open HelloThere
   - add an empty caption field
 - Go to the sidebar Content
 - And now as a newbie user try to find the problem, which causes the empty TOC line.

Reply:
That's the rationale I was expecting, but my point above is that having an empty caption field is a mistake that will cause problems anywhere that the caption field is used. Having special logic to "fix" things in the special case of the TOC (but not others) actually makes the system more complex overall.

Reply:
I'm OK with your reasoning. I did respond in the same way with [my first post](https://github.com/Jermolene/TiddlyWiki5/issues/3624#issuecomment-444930269)

So I'll restore the "existing" behaviour